### PR TITLE
Test cases list creation

### PR DESCRIPTION
**Pull request by monikasmolarek**
I created a doc covering the test cases that should be covered while testing the lite wallet manually. 
It aims to help remember what and how to tests, but also to track the results.

For now only 3 main processes starting from the main page.
The wallet screen and all the components, options, and settings will be added soon.

### CODE REVIEW

File: TestCases.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+9. Ensure that only one word can be placed in one slot at a time. If there was already a word, it should pop out to the initial top position above the grid.
+10. Check that a word, once placed, remains in the slot unless dragged out again.
+11. Ensure the user can easily drag the word out of the incorrect slot and try again.
+12. Verify that correctly placing the words into their slots provides visual feedback.
+13. Ensure that an error message or visual indication is provided if the words are placed in the wrong slots.
+
+**Back button**
+
+1. Verify that clicking "Back" navigates the user back to the mnemonic phrase page.
+2. Ensure that the mnemonic phrase is retained when returning to the mnemonic phrase page.
```
Comment (dioexul) [inline TestCases.md:62]: I would add 
3. Ensure that clicking 'Back' and after 'Next Step' clears all filled words and prompts the user to fill in the 4 words again.

File: TestCases.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+4. Ensure the 4 missing words are listed above the grid for drag-and-drop.
+5. Verify the presence of a "Back" button.
+6. Verify the presence of a "Next Step" button.
+7. Verify that the 4 missing words can be dragged and dropped into the empty slots on the grid.
+8. Ensure that each word can only be placed in one slot.
+9. Ensure that only one word can be placed in one slot at a time. If there was already a word, it should pop out to the initial top position above the grid.
+10. Check that a word, once placed, remains in the slot unless dragged out again.
+11. Ensure the user can easily drag the word out of the incorrect slot and try again.
+12. Verify that correctly placing the words into their slots provides visual feedback.
+13. Ensure that an error message or visual indication is provided if the words are placed in the wrong slots.
```
Comment (dioexul) [inline TestCases.md:57]: == the button "next" is inactive 
10.5 ensure that visually all words inserted in the same way (with no additional space or changing sizes)

File: TestCases.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+1. Verify that clicking "Back" navigates the user back to the mnemonic phrase page.
+2. Ensure that the mnemonic phrase is retained when returning to the mnemonic phrase page.
+3. Ensure that clicking 'Back' clears all filled words and prompts the user to fill in the 4 words again.
+
+**Next step button**
+
+1. Verify that clicking "Next Step" is only enabled when all 4 words are correctly placed in the grid.
+2. Ensure that clicking "Next Step" navigates the user to the next page i.e. password setup.
+3. Check behavior when attempting to click "Next Step" with only some of the words correctly placed.
+4. Ensure that clicking 'Next Step' clears all filled words and prompts the user to fill in the 4 words again.
```
Comment (dioexul) [inline TestCases.md:70]: This one from the previous step (here item 2. is correct)

File: TestCases.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+7. Check that the page displays: "Back" button.
+8. Check that the page displays: Green "Create Wallet" button.
+9. Verify clicking the eye icon reveals the entered password in the "Set the password:" field.
+10. Verify clicking the eye icon reveals the entered password in the "Confirm the password:" field.
+11. Ensure clicking the eye icon again hides the password.
+12. Verify that the "Confirm the password:" field must match the "Set the password:" field.
+13. Verify an appropriate error message is shown if the passwords in the "Set the password" and "Confirm the password" fields do not match.
+14. Verify the system correctly handles and accepts special characters in the password fields.
+15. Verify that the password is securely transmitted and not exposed in any way.
+16. Ensure the app correctly handles password encryption and validation.
```
Comment (dioexul) [inline TestCases.md:89]: 17. Ensure that appropriate error messages is shown when a password doesn't meet the requirements (now it is "Password should contain symbols in upper and lower cases, numbers, and special characters"). 
Ideally to add 18. Ensure that when the focus is on the field the suggestion pop-up is shown - what should password contain

File: TestCases.md [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+24. Ensure only the last selected file is processed, or an appropriate message is shown if multiple files are not supported.
+25. Verify that the system provides immediate feedback upon correct file selection.
+26. Verify that the user can remove or change the selected file before entering the password.
+27. Ensure the system correctly updates the file selection and resets any error messages after retrying.
+28. Verify that users can cancel the file upload process if needed.
+29. Ensure that any partially uploaded files are handled correctly and the user can retry.
+
+# Log in / Log out
+
+1. Verify that navigating to `https://wallet.spacemesh.io/` redirects instantly to `https://wallet.spacemesh.io/unlock`.
```
Comment (dioexul) [inline TestCases.md:177]: One of the last commits changed it. Also it should redirect only after reload page, not all the time.
Like https://wallet.spacemesh.io/#/ shall redirect to https://wallet.spacemesh.io/#/wallet

File: TestCases.md (hunk 2 above)
Comment (monikasmolarek) [reply TestCases.md:57]: added, thank you

File: TestCases.md (hunk 5 above)
Comment (monikasmolarek) [reply TestCases.md:177]: This is specifically about the log in functionality, very narrow. Not about the page reloads and redirects in general. It's only about whether the app behaves correctly when we want to log in to the wallet, so the moment that we have our wallet set up and we open the browser and want to log in. So if someone goes to /#/ or /#/wallet, he should be redirected to login.
I will reformulate it as follows:

```suggestion
1.  With a wallet correctly set up, and logged out, verify that navigating to any valid link`https://wallet.spacemesh.io/#/[any-existing-path]` redirects instantly to `https://wallet.spacemesh.io/#/unlock`.
```

**Comment by monikasmolarek** [comment]
@brusherru I will let myself to merge this doc. You can of course move, remove It, do whatever you want. But It would probably make it easier to review the next pull request, as Yulia has already reviewed these ones. 
This list is not crucial, but might make it easier to document wallet and test it. I think of using it as a basis to write automatic integration/end-to-end tests in the future. It doesn’t break anything in the code, so merging for now here. I will reorganize it better when it's more robust.

**Pull request merged**